### DISCUSSION TITLE
关于RoutingObject中的domainStrategy有一些疑问

### DISCUSSION THREAD

**Discussion by slinar** [XTLS/Xray-core#3378] [Q&A]
https://xtls.github.io/config/routing.html 中的文档所描述：
> "AsIs"：只使用域名进行路由选择。默认值。（言下之意：跳过IP规则匹配）

https://xtls.github.io/config/inbound.html#sniffingobject 提示中所描述：
> 在能保证 被代理连接能得到正确的 DNS 解析 时，使用 routeOnly 且开启 destOverride 的同时，将路由匹配策略 domainStrategy 设置为 AsIs 即可实现全程无 DNS 解析进行域名及 IP 分流。此时遇到 IP 规则匹配时使用的 IP 为域名原始 IP。

也就是说路由规则中使用"AsIs"还是会进行IP规则的匹配？或者是有其他意义？

**Comment by Fangliding** [comment]
em ASIS 顾名思义应该是 是什么就是什么 如果是域名就是用域名 不再去解析IP之类的乱七八糟的操作
如果客户端发来的请求就是指向IP的 那肯定还是得匹配IP规则的

**Reply by wtfr-dot** [reply]
大佬，我的理解是这样的，不知对不对，请指点！
客户端访问目标网站都是通过ip进行连接的，domainStrategy 只对域名生效，我觉得它最重要的是对域名解析的分流，客户端向目标发起连接前如果域名没有解析过，这时首先进行域名解析，这时ip包里目标ip一般是网关，数据里就含有要解析的域名，进入xray后被嗅探出来进行路由决策，如果是ASIS，那它就根据geosite等预置的域名规则匹配，匹配不上就通过outbound的第一个节点发出去，有可能是去远程vps解析dns，总之最后得到ip，然后客户端根据得到的ip地址，再向目标发起连接，ip包里这时的目标ip就是真正需要连接的目标了，进入xray后再次通过路由决策时就不需要进行域名匹配了，而是geoip这类的ip规则，从而完成访问目标的分流
